sdktests/server_side_events_summary.go

```diff
@@ -0,0 +1,277 @@
+package sdktests
+
+import (
+	"github.com/launchdarkly/sdk-test-harness/framework/ldtest"
+	"github.com/launchdarkly/sdk-test-harness/mockld"
+	"github.com/launchdarkly/sdk-test-harness/servicedef"
+
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/lduser"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldvalue"
+	"gopkg.in/launchdarkly/go-server-sdk-evaluation.v1/ldbuilders"
+	"gopkg.in/launchdarkly/go-server-sdk-evaluation.v1/ldmodel"
+)
+
+func doServerSideSummaryEventTests(t *ldtest.T) {
+	t.Run("basic counter behavior", doServerSideSummaryEventBasicTest)
+	t.Run("unknown flag", doServerSideSummaryEventUnknownFlagTest)
+	t.Run("reset after each flush", doServerSideSummaryEventResetTest)
+	t.Run("prerequisites", doServerSideSummaryEventPrerequisitesTest)
+}
+
+func doServerSideSummaryEventBasicTest(t *ldtest.T) {
+	flag1 := ldbuilders.NewFlagBuilder("flag1").Version(100).
+		Variations(ldvalue.String("value1a"), ldvalue.String("value1b")).
+		On(true).FallthroughVariation(0).
+		AddTarget(1, "user-b").
+		Build()
+
+	flag2 := ldbuilders.NewFlagBuilder("flag2").Version(200).
+		Variations(ldvalue.String("value2a"), ldvalue.String("value2b")).
+		On(true).FallthroughVariation(0).
+		AddTarget(1, "user-b").
+		Build()
+
+	userA := lduser.NewUser("user-a")
+	userB := lduser.NewUser("user-b")
+	default1 := ldvalue.String("default1")
+	default2 := ldvalue.String("default2")
+
+	dataBuilder := mockld.NewServerSDKDataBuilder()
+	dataBuilder.Flag(flag1, flag2)
+
+	dataSource := NewSDKDataSource(t, dataBuilder.Build())
+	events := NewSDKEventSink(t)
+	client := NewSDKClient(t, dataSource, events)
+
+	// evaluations for flag1: two for userA producing value1a, one for userB producing value1b
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: &userA, DefaultValue: default1})
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: &userB, DefaultValue: default1})
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag1.Key, User: &userA, DefaultValue: default1})
+
+	// evaluations for flag2: one for userA producing value2a
+	_ = client.EvaluateFlag(t, servicedef.EvaluateFlagParams{FlagKey: flag2.Key, User: &userA, DefaultValue: default2})
+
+	client.FlushEvents(t)
+	payload := events.ExpectAnalyticsEvents(t, defaultEventTimeout)
+
+	m.In(t).Assert(payload, m.Items(
+		EventIsIndexEvent(mockld.SimpleEventUser(userA)),
+		EventIsIndexEvent(mockld.SimpleEventUser(userB)),
+		m.JSONMap().Should(m.MapOf(
```

Comment:
This kind of nested map assertion is a feature that was added in https://github.com/launchdarkly/go-test-helpers/pull/12. It's different from just doing a deep-equality test on a map, because you can nest other kinds of assertions inside it (like `ItemsInAnyOrder`).